server/util/healthcheck/healthcheck.go

```diff
@@ -109,6 +112,41 @@ func (h *HealthChecker) Statusz(ctx context.Context) string {
 	return buf
 }
 
+func (h *HealthChecker) handleSignals(signalChan <-chan os.Signal) {
+	// When running in a terminal, the ^C character echoed back to the user
+	// messes up the log output a bit. So, print a newline every time we get
+	// a signal to make the output a little cleaner.
+	isTTY := isatty.IsTerminal(uintptr(os.Stderr.Fd()))
+	sig := <-signalChan
+	if isTTY {
+		fmt.Println()
+	}
+	log.Infof("Caught %s signal; starting graceful shutdown (hard-stopping in %s)", sig, *maxShutdownDuration)
+	hardStopTime := time.Now().Add(*maxShutdownDuration)
+	h.Shutdown()
+	// If we keep getting SIGINT/SIGTERM, the user is probably mashing
+	// Ctrl+C and really wants to kill the server. After 3 signals, exit
+	// immediately. Before then, report the current status so the user knows
+	// we got their request but are just still shutting down.
+	n := 0
+	const ignoreCount = 3
```

Comment:
Are there any production situations where we could receive multiple SIGINTs or SIGTERMs and still want to shutdown cleanly?

Reply:
Good question; I'm not sure, so I changed the logic so we only do the `log.Fatal` when running in a TTY.